### ISSUE TITLE
Key Vault set-policy needs "--applicationId" parameter for OBO Token (user+application)

### ISSUE THREAD

**Issue by mattsmall1972** [Azure/azure-cli#9034]
**Describe the bug**
az keyvault set-policy needs an additional parameter to allow users to add an on-behalf-of access policy for user+application. 

**To Reproduce**
No reproduction needed.  It simply does not exist.

**Expected behavior**
Add "--applicationId" as a parameter which populates the "applicationId" property of an Azure Key Vault.  This feature exists in the Azure Portal as well as Powershell and REST APIs.

**Environment summary**
All AFAIK.

**Additional context**
[Documenation to -ApplicationId for Set-AzKeyVaultAccessPolicy](https://docs.microsoft.com/en-us/powershell/module/az.keyvault/set-azkeyvaultaccesspolicy?view=azps-1.7.0#parameters)

**Comment by Magn3s1um** [comment]
You can circumnavigate this by using the ApplicationId of your application as the ServicePrincipalName parameter.  Confusing?  Yes.

**Comment by mattsmall1972** [comment]
That's not a valid workaround.  Your suggestion does not fix the issue as the value of the ApplicationId is required for OBO tokens.

**Comment by yonzhan** [comment]
add to S164.

**Comment by bim-msft** [comment]
@mattsmall1972 Hi, are you still being blocked by this issue? Does `az keyvault set-policy --object-id` work for your case? Thanks.

**Comment by yungezz** [comment]
hi @mattsmall1972 is this still an issue for you? Does `az keyvault set-policy --object-id` work for you? We'll close the issue for now. Feel free to create new issue if there's any. thanks.
